skops/io/_persist.py

```diff
@@ -15,7 +15,7 @@
 # We load the dispatch functions from the corresponding modules and register
 # them. Old protocols are found in the 'old/' directory, with the protocol
 # version appended to the corresponding module name.
-modules = ["._general", "._numpy", "._scipy", "._sklearn"]
+modules = ["._general", "._numpy", "._scipy", "._sklearn", "._quantile_forest"]
```

Comment:
Right, good catch. What I think we should do is to delay required import to when the user dumps, at which point if they have an instance from this library, they should have the library installed. And then when loading, the same, we try to load the module / type, and if it doesn't exist we raise.

Reply:
Is the issue not more during loading though? We need to import the class because `get_state` dispatches based on it. Therefore, we cannot lazily load `QuantileForest` when the object is constructed. Otherwise, I would agree.

Reply:
Loading depends on dispatch based on the dictionaries we're setting based on the `__loader__`, not type, so that dispatch is fine and can delay any imports to when they're needed, I think?

Reply:
Ah yes, sorry, you're right, it's needed during dumping, not loading. So is your suggestion that this code:

https://github.com/skops-dev/skops/blob/923dafb64433f599b2f62f8b746938979e1aea03/skops/io/_persist.py#L20-L26

should not be called on the module root, but instead the first time that `dump` / `dumps` is being called?

Reply:
what we could do, is inside `_quantile_forest.py` only do lazy imports, and only fill the dictionary if the library is installed, that should fix the issue I think.

Reply:
That should work too.

Reply:
Thanks for the conversation; updated with lazy imports inside `_quantile_forest.py`.